### ISSUE TITLE
it halt when applicating with ida7.6 sp1

### ISSUE THREAD

**Issue by liquan165** [binaryai/sdk#117]
After draging any binary ,error occrred.

**Comment by liquan165** [comment]
![image](https://user-images.githubusercontent.com/16474163/129511202-c8603c37-dab7-4fcc-8c1d-8b0d6ca954be.png)

**Comment by nforest** [comment]
Currently I don't have a IDA7.6SP1 in my hand. however, I tested the plugin on IDA7.6.210427 and it works for me.
It's rare that a pure python plugin can make IDA report internal error(sth in the snapshot you uploaded).
Hope that you can help reproduce the bug in a clean environment(pip install binaryai, and without any other 3rd-party plugins), then report the detailed error log to us.
Thanks!

**Comment by liumuqing** [comment]
@nforest  I can reproduce a similar bug on IDA 7.7, I don't have IDA 6.6 SP1 neither...


the internal error 2028 of @liquan165 can be fixed by
replace the line at https://github.com/binaryai/sdk/blob/v0.2.8/binaryai/ida.py#L14 to 
```python3
import lazy_object_proxy
All_STR = lazy_object_proxy.Proxy(lambda: dict(map(lambda i: (i.ea, str(i)), idautils.Strings())))
```
seems Strings is not initialized when binaryai is imported?


after appling the lazy_object_proxy patch, a different error appers on IDA 7.7
![图片](https://user-images.githubusercontent.com/1625020/148150727-d94e3962-f7fb-42bf-9bf9-7da4e25213bf.png)

**Comment by liumuqing** [comment]
@nforest  I can reproduce a similar bug on IDA 7.7, I don't have IDA 6.6 SP1 either...


the internal error 2028 of @liquan165 can be fixed by
replace the line at https://github.com/binaryai/sdk/blob/v0.2.8/binaryai/ida.py#L14 to 
```python3
import lazy_object_proxy
All_STR = lazy_object_proxy.Proxy(lambda: dict(map(lambda i: (i.ea, str(i)), idautils.Strings())))
```
seems Strings is not initialized when binaryai is imported?



but after appling the lazy_object_proxy patch, a different error appears
![图片](https://user-images.githubusercontent.com/1625020/148153245-0d064c12-65a6-44ee-a055-9a333c46575a.png)

**Comment by liumuqing** [comment]
@nforest this patch enable binaryai to run on IDA7.7
https://github.com/binaryai/sdk/compare/binaryai:v0.2.8...liumuqing:v0.2.8_ida7.7_fix?expand=1

**Comment by davendu** [comment]
After the overall refactor in Q1 2023, the customized function matching feature have been removed from the server side. As the result, this SDK is deprecated, so the issue will be closed (as well as closing PR #120).

To perform function matching now, try upload your file to our online service, and [our new plugin](https://github.com/binaryai/plugins) to load the result.
